### DISCUSSION TITLE
Using R1/R2 instead of F1/F2

### DISCUSSION THREAD

**Discussion by Code394** [omerfaruk-aran/esphome_samsung_hvac_bus#185] [General]
I have a system of 21 NASA units connected through an F1/F2 interface. A cable from R1/R2 setup runs to my server, where I use S-NET Pro 2 software to diagnose and manage faults in air conditioners.

The software allows me to select whether I'm connected via R1/R2 or F1/F2. However, in my case, I'm always physically connected to R1/R2. Despite this, the system seems to function the same way—whether I select R1/R2 or F1/F2 in the software—providing the same information and allowing me to change the air conditioners' statuses (except for programming codes).

Given that everything works identically, regardless of whether I choose R1/R2 or F1/F2 in the software, is there any technical reason why Won't it work on R1/R2 too?

**Comment by lanwin** [comment]
Hi @htht2001, no technical reason. From what I saw in the SNET code R1/R2 also uses RS485 and also uses the NASA protocol. But it seems some commands are different. But it should be possible to make it work.

**Comment by Code394** [comment]
In the end I used F1/F2
R1/R2 should be slower

**Comment by Code394** [comment]
In the end I can state that R1/R2 will also work equally well (except for the possibility to change programming codes)
But in normal cases there is no reason to use it because it works at a slower rate
This is only useful if there is more than one gas circuit
(because then there is a separate F1/F2 for each circuit)
So to control all the circuits you have to use R1/R2
F3/F4 really works differently and won't work
I created a pull request [186](https://github.com/lanwin/esphome_samsung_ac/pull/186)
to add information about R1/R to readme.md
